nomad/drainer/drainer.go

```diff
@@ -399,11 +400,12 @@ func (n *NodeDrainer) drainAllocs(future *structs.BatchFuture, allocs []*structs
 	var finalIndex uint64
 	for _, u := range partitionAllocDrain(transistions, evals) {
 		index, err := n.raft.AllocUpdateDesiredTransition(u.Transistions, u.Evals)
+	var mErr multierror.Error
 		if err != nil {
-			future.Respond(index, err)
+			mErr.Errors = append(mErr.Errors, err)
```

Comment:
Revert back to old single error and do a return after Responding. Don't need to keep attempting the Raft commit as it is likely from a loss of leadership.